### DISCUSSION TITLE
RP2040 support

### DISCUSSION THREAD

**Discussion by glennswest** [khoih-prog/AsyncHTTPSRequest_Generic#9] [General]
Looking at RP2040, be nice to have this work there. I could do up to 8 channels for my gateway.
32 bit, and some have on board wifi.

**Comment by khoih-prog** [comment]
The obstacle for RP2040 and many other boards, using W5x00 Ethernet or WiFi (NINA, etc.), is the Ethernet/WiFi libraries are not designed to support Async yet.

Only some libraries, based on LwIP stack, are supporting now, such as

1. Teensy 4.1 **QNEthernet**
2. ESP32 WiFi and Ethernet (WT32_ETH01). This is the **promising** solution for you.
3. STM32 Ethernet using LAN8742A or LAN8720
4. Portenta_H7 Ethernet/WiFi

We certainly have to wait for someone to rewrite those libraries as I don't have enough time / knowledge to finish all of the tasks.
Any help will be much appreciated by all of us.

**Reply by khoih-prog** [reply]
RP2040W using CYW43439 WiFi, using [**arduino-pico** core v2.5.0+](https://github.com/earlephilhower/arduino-pico) can be a candidate.

Will spend some time to look at it to see if ready / easy or not.

RP2040 using the same core and lwip-based Ethernet, such as W5x00, ENC28J60, could also be considered whenever this enhancement [[**Feature Request] Port these libraries lwIP_w5100, lwIP_w5500 and lwIP_enc28j60 from ESP8266** #775](https://github.com/earlephilhower/arduino-pico/issues/775) done.
